scripts/sanity_chk/sanitylib.py

```diff
@@ -1612,78 +1612,75 @@ def __init__(self, testcase, platform, outdir):
         self.name = os.path.join(platform.name, testcase.name)
         self.build_dir = os.path.join(outdir, platform.name, testcase.name)
 
-        self.build_only = True
         self.run = False
 
         self.results = {}
 
     def __lt__(self, other):
         return self.name < other.name
 
+
+    def testcase_runnable(self, testcase, fixtures):
+        can_run = False
+        # console harness allows us to run the test and capture data.
+        if testcase.harness in [ 'console', 'ztest']:
+
+            # if we have a fixture that is also being supplied on the
+            # command-line, then we need to run the test, not just build it.
+            fixture = testcase.harness_config.get('fixture')
+            if fixture:
+                if fixture in fixtures:
+                    can_run = True
+                else:
+                    can_run = False
+            else:
+                can_run = True
+
+        elif testcase.harness:
+            can_run = False
+        else:
+            can_run = True
+
+        return can_run
+
+
     # Global testsuite parameters
     def check_build_or_run(self, build_only=False, enable_slow=False, device_testing=False, fixtures=[]):
 
```

Comment:
Maybe you can consider adding a description here? The name is `check_build_or_run` but the function returns boolean, so it is not obvious at the first glance that `False` = build and `True` = run (or in fact build and run as they are not exclusive). The logic looks clear but the name can be confusing. Or the name could be self-explanatory if it is changed to sth like `check_if_runnable' or even just `runnable` as this is called as `instance.runnable()` and above you define `testcase_runnable` which also returns boolean